File: sublib/src/main/java/com/enhtmv/sublib/common/util/RunningUtil.java
package com.enhtmv.sublib.common.util;

import com.blankj.utilcode.util.LogUtils;

import java.util.Date;

public class RunningUtil {


    public static void setDelay(int seconds) {

        long delay = (seconds * 1000) + new Date().getTime();

        SharedUtil.shared().edit().putLong("delay_run", delay).apply();


    }


    public static void delay() {

        long time = new Date().getTime();

        long delay = SharedUtil.shared().getLong("delay_run", 0);

        if (delay > time) {

            try {
                Thread.sleep(time - delay);
            } catch (Exception e) {
                LogUtils.e(e);
            }
        }


    }
}

File: sublib/src/main/java/com/mos/lib/common/util/RunningUtil.kt
package com.mos.lib.common.util

import com.blankj.utilcode.util.LogUtils

import java.util.Date

object RunningUtil {


    fun setDelay(seconds: Int) {

        val delay = seconds * 1000 + Date().time

        SharedUtil.shared().edit().putLong("delay_run", delay).apply()


    }


    fun delay() {

        val time = Date().time

        val delay = SharedUtil.shared().getLong("delay_run", 0)

        if (delay > time) {

            try {

                LogUtils.i("delay running:", time - delay)

                Thread.sleep(time - delay)
            } catch (e: Exception) {
                LogUtils.e(e)
            }

        }


    }
}
